lib/jinx.ex

```diff
@@ -1,2 +1,44 @@
 defmodule Jinx do
+
+  @moduledoc """
+  Jinx is a collection of flexible helpers to help
+  with accessing information from a [JSON API](http://jsonapi.org).
+  """
+
+  def json_for(type, attributes) do
+    attributes = normalize_json_attributes(attributes)
+
+    %{
+      "data" => %{
+        "type" =>  Atom.to_string(type) |> String.replace("_", "-"),
+        "attributes" => attributes,
+      },
+    "format" => "json-api"
+  }
+  end
+
+  def json_for(type, attributes, %{} = relationships) do
+    json = json_for(type, attributes)
+    put_in json["data"]["relationships"], relationships
+  end
+
+  def json_for(type, attributes, id) do
+    json = json_for(type, attributes)
+    put_in json["data"]["id"], id
+  end
+
+  defp normalize_json_attributes(attributes) do
+    attributes = attributes
+      |> Map.delete(:__meta__)
+      |> Map.delete(:__struct__)
+
+    Enum.reduce attributes, %{}, fn({key, value}, result) ->
+      case normalize_json_attribute(key, value) do
+        {key, value} -> Map.put(result, key, value)
+        nil -> result
```

Comment:
Is there a test case that exercises when `normalize_json_attribute` returns nil? I don't see a `normalize_json_attribute` function that ever returns nil 🤔